components/RatingStar.js

```diff
@@ -0,0 +1,41 @@
+import styled from "styled-components";
+import { useState } from "react";
+
+const StarRating = ({ rating, onRate }) => {
+  const [hoverRating, setHoverRating] = useState(0);
+
+  return (
+    <Stars>
+      {[1, 2, 3, 4, 5].map((star) => (
+        <Star
+          key={star}
+          filled={star <= (hoverRating || rating)}
```

Comment:
When passing props to a styled component you should prefix them with "$". Otherwise you will get a message in your terminal:


> it looks like an unknown prop "filled" is being sent through to the DOM, which will likely trigger a React console error. If you would like automatic filtering of unknown props, you can opt-into that behavior via `<StyleSheetManager shouldForwardProp={...}>` (connect an API like `@emotion/is-prop-valid`) or consider using transient props (`$` prefix for automatic filtering.)

```suggestion
          $filled={star <= (hoverRating || rating)}
```